lib/rails_erd/domain.rb

```diff
@@ -154,7 +155,11 @@ def check_polymorphic_association_validity(association)
       entity = entity_by_name(entity_name)
 
       if entity || (entity && entity.generalized?)
-        return entity
+        if check_model_validity(entity)
+          return entity
+        else
+          raise("polymorphic entity #{entity_name} is not a valid model")
+        end
```

Comment:
> Honestly, I'm puzzled right this moment about what this line is doing:

yup. was thinking about making a PR for that too. the `||` portion of it can never get exercised. the problem is, like you, I have no idea what the 2nd half was intended to do. There's obviously no test for it (or else it'd be failing). On the other hand, does it matter if no-one is complaining? Probably...